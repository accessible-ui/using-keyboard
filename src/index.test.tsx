import * as React from 'react'
import {render} from '@testing-library/react'
import {UsingKeyboard, BodyUsingKeyboard} from './index'

test('UsingKeyboard -> adds classname', () => {
  expect(
    render(
      <UsingKeyboard defaultUsingKeyboard={false}>
        <span>Foo</span>
      </UsingKeyboard>
    ).asFragment()
  ).toMatchSnapshot('without class name')

  expect(
    render(
      <UsingKeyboard defaultUsingKeyboard>
        <span>Foo</span>
      </UsingKeyboard>
    ).asFragment()
  ).toMatchSnapshot('with class name')

  expect(
    render(
      <UsingKeyboard defaultUsingKeyboard={false}>
        <span className='foo'>Foo</span>
      </UsingKeyboard>
    ).asFragment()
  ).toMatchSnapshot('without class name appended')

  expect(
    render(
      <UsingKeyboard defaultUsingKeyboard>
        <span className='foo'>Foo</span>
      </UsingKeyboard>
    ).asFragment()
  ).toMatchSnapshot('with class name appended')

  expect(
    render(
      <UsingKeyboard className='keyboarding' defaultUsingKeyboard>
        <span>Foo</span>
      </UsingKeyboard>
    ).asFragment()
  ).toMatchSnapshot('with custom class')
})

test('<BodyUsingKeybooard> -> adds classname', () => {
  expect(
    render(<BodyUsingKeyboard defaultUsingKeyboard={false} />).baseElement
  ).toMatchSnapshot('without class name')

  expect(
    render(<BodyUsingKeyboard defaultUsingKeyboard />).baseElement
  ).toMatchSnapshot('with class name')

  document.body.removeAttribute('class')
  document.body.setAttribute('class', 'foo')

  expect(
    render(<BodyUsingKeyboard defaultUsingKeyboard={false} />).baseElement
  ).toMatchSnapshot('without class name appended')

  expect(
    render(<BodyUsingKeyboard defaultUsingKeyboard />).baseElement
  ).toMatchSnapshot('with class name appended')

  document.body.removeAttribute('class')

  expect(
    render(<BodyUsingKeyboard defaultUsingKeyboard className='keyboarding' />)
      .baseElement
  ).toMatchSnapshot('with custom class')

  document.body.removeAttribute('class')

  expect(
    render(<BodyUsingKeyboard>Foo</BodyUsingKeyboard>).baseElement
  ).toMatchSnapshot('with children')
})
