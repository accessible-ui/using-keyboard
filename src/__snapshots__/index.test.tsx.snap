// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyUsingKeybooard> -> adds classname: with children 1`] = `
<body>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div>
    Foo
  </div>
</body>
`;

exports[`<BodyUsingKeybooard> -> adds classname: with class name 1`] = `
<body
  class="using-keyboard"
>
  <div />
  <div />
</body>
`;

exports[`<BodyUsingKeybooard> -> adds classname: with class name appended 1`] = `
<body
  class="foo using-keyboard"
>
  <div />
  <div />
  <div />
  <div />
</body>
`;

exports[`<BodyUsingKeybooard> -> adds classname: with custom class 1`] = `
<body
  class="keyboarding"
>
  <div />
  <div />
  <div />
  <div />
  <div />
</body>
`;

exports[`<BodyUsingKeybooard> -> adds classname: without class name 1`] = `
<body>
  <div />
</body>
`;

exports[`<BodyUsingKeybooard> -> adds classname: without class name appended 1`] = `
<body
  class="foo"
>
  <div />
  <div />
  <div />
</body>
`;

exports[`UsingKeyboard -> adds classname: with class name 1`] = `
<DocumentFragment>
  <span
    class="using-keyboard"
  >
    Foo
  </span>
</DocumentFragment>
`;

exports[`UsingKeyboard -> adds classname: with class name appended 1`] = `
<DocumentFragment>
  <span
    class="foo using-keyboard"
  >
    Foo
  </span>
</DocumentFragment>
`;

exports[`UsingKeyboard -> adds classname: with custom class 1`] = `
<DocumentFragment>
  <span
    class="keyboarding"
  >
    Foo
  </span>
</DocumentFragment>
`;

exports[`UsingKeyboard -> adds classname: without class name 1`] = `
<DocumentFragment>
  <span
    class=""
  >
    Foo
  </span>
</DocumentFragment>
`;

exports[`UsingKeyboard -> adds classname: without class name appended 1`] = `
<DocumentFragment>
  <span
    class="foo"
  >
    Foo
  </span>
</DocumentFragment>
`;
